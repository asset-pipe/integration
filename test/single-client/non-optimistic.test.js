'use strict';

const { resolve } = require;
const { spawn } = require('child_process');
const Client = require('@asset-pipe/client');
const buildServerUri = 'http://127.0.0.1:7205';
const client = new Client({ buildServerUri });
const supertest = require('supertest');
const request = supertest(buildServerUri);
const prettier = require('prettier');

const localiseBodyPaths = body =>
    JSON.parse(
        JSON.stringify(body, null, 2).replace(
            /"file":\s".*\/asset-pipe\//g,
            '"file": "'
        )
    );

let server;

function startServer(env = 'development') {
    return new Promise((resolve, reject) => {
        server = spawn('./node_modules/.bin/asset-pipe-server', [], {
            env: Object.assign({}, process.env, {
                NODE_ENV: env,
                PORT: 7205,
            }),
        });
        server.stdout.once('data', () => resolve());
        server.stderr.once('data', () => resolve());
        server.once('error', err => reject(err));
    });
}

function stopServer() {
    return new Promise(resolve => {
        server.once('close', () => resolve());
        server.kill();
    });
}

beforeAll(() => {
    jest.setTimeout(20000);
    return startServer();
});

afterAll(() => stopServer());

test('Client uploads a js feed to build server', async () => {
    expect.assertions(1);
    const result = await client.uploadFeed([resolve('../../assets/index.js')]);
    expect(result).toMatchSnapshot();
});

test('Client gets uploaded js feed from build server', async () => {
    expect.assertions(1);
    const result = await client.uploadFeed([resolve('../../assets/index.js')]);
    const { body } = await request
        .get(`/feed/${result.file}`)
        .set('Accept', 'application/json')
        .expect('Content-Type', /application\/json/)
        .expect(200);
    expect(localiseBodyPaths(body)).toMatchSnapshot();
});

test('Client uploads a css feed to build server', async () => {
    expect.assertions(1);
    const result = await client.uploadFeed([resolve('../../assets/style.css')]);
    expect(result).toMatchSnapshot();
});

test('Client gets uploaded css feed from build server', async () => {
    expect.assertions(1);
    const result = await client.uploadFeed([resolve('../../assets/style.css')]);
    const { body } = await request
        .get(`/feed/${result.file}`)
        .set('Accept', 'application/json')
        .expect('Content-Type', /application\/json/)
        .expect(200);
    expect(body).toMatchSnapshot();
});

test('Client requests bundling of a js feed on build server', async () => {
    expect.assertions(1);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/index.js'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'js'
    );
    expect(bundleResponse).toMatchSnapshot();
});

test('Client gets bundled js file from build server', async () => {
    expect.assertions(1);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/index.js'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'js'
    );
    const { text } = await request
        .get(`/bundle/${bundleResponse.file}`)
        .expect(200);
    expect(text).toMatchSnapshot();
});

test('Client requests bundling of a css feed on build server', async () => {
    expect.assertions(1);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/style.css'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'css'
    );
    expect(bundleResponse).toMatchSnapshot();
});

test('Client gets bundled css file from build server', async () => {
    expect.assertions(1);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/style.css'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'css'
    );
    const { text } = await request
        .get(`/bundle/${bundleResponse.file}`)
        .expect(200);
    expect(text).toMatchSnapshot();
});

test('Uploaded js feed from build server is not deduped', async () => {
    expect.assertions(1);
    const result = await client.uploadFeed([
        resolve('../../assets/duped-before-uploading.js'),
    ]);
    const { body } = await request
        .get(`/feed/${result.file}`)
        .set('Accept', 'application/json')
        .expect('Content-Type', /application\/json/)
        .expect(200);
    expect(localiseBodyPaths(body)).toMatchSnapshot();
});

test('Bundled js feed from build server is deduped', async () => {
    expect.assertions(1);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/duped-before-uploading.js'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'js'
    );
    const { text } = await request
        .get(`/bundle/${bundleResponse.file}`)
        .expect(200);
    expect(text).toMatchSnapshot();
});

test('NODE_ENV variables replaced with hard coded values', async () => {
    expect.assertions(5);
    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/node_envs.js'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'js'
    );
    const { text } = await request
        .get(`/bundle/${bundleResponse.file}`)
        .expect(200);
    expect(text).not.toMatch(`process.env.NODE_ENV === 'development'`);
    expect(text).not.toMatch(`process.env.NODE_ENV === 'production'`);
    expect(text).toMatch(`"development" === 'production'`);
    expect(text).toMatch(`"development" === 'development'`);
    expect(prettier.format(text)).toMatchSnapshot();
});

test('NODE_ENV variables trigger dead code elimination (DCE)', async () => {
    expect.assertions(4);
    await stopServer();
    await startServer('production');

    const uploadResponse = await client.uploadFeed([
        resolve('../../assets/node_envs.js'),
    ]);
    const bundleResponse = await client.createRemoteBundle(
        [uploadResponse.file],
        'js'
    );
    const { text } = await request
        .get(`/bundle/${bundleResponse.file}`)
        .expect(200);
    expect(text).toMatch(`In prod!!`);
    expect(text).not.toMatch(`In dev!!`);
    expect(text).not.toMatch(`In limbo!!`);
    expect(text).toMatchSnapshot();
    await stopServer();
    await startServer();
});
