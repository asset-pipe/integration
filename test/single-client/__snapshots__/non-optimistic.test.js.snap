// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bundled js feed from build server is deduped 1`] = `
"(function(){function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require==\\"function\\"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error(\\"Cannot find module '\\"+o+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require==\\"function\\"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s}return e})()({1:[function(require,module,exports){
'use strict';

const sum = require('./d');

sum(1, 2);

},{\\"./d\\":2}],2:[function(require,module,exports){
'use strict';

module.exports = function sum(a, b) {
    return a + b;
};

},{}],3:[function(require,module,exports){
require('./a');

},{\\"./a\\":1}],4:[function(require,module,exports){
arguments[4][3][0].apply(exports,arguments)
},{\\"./a\\":1,\\"dup\\":3}],5:[function(require,module,exports){
arguments[4][3][0].apply(exports,arguments)
},{\\"./a\\":1,\\"dup\\":3}],6:[function(require,module,exports){
const a = require('./dup-1');
const b = require('./dup-2');
const c = require('./dup-3');

},{\\"./dup-1\\":3,\\"./dup-2\\":4,\\"./dup-3\\":5}]},{},[6]);
"
`;

exports[`Client gets bundled css file from build server 1`] = `
"html, body, div, span, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
abbr, address, cite, code,
del, dfn, em, img, ins, kbd, q, samp,
small, strong, sub, sup, var,
b, i,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section, summary,
time, mark, audio, video {
    margin:0;
    padding:0;
    border:0;
    outline:0;
    font-size:100%;
    vertical-align:baseline;
    background:transparent;
}

body {
    line-height:1;
}

article,aside,details,figcaption,figure,
footer,header,hgroup,menu,nav,section { 
    display:block;
}

nav ul {
    list-style:none;
}

blockquote, q {
    quotes:none;
}

blockquote:before, blockquote:after,
q:before, q:after {
    content:'';
    content:none;
}

a {
    margin:0;
    padding:0;
    font-size:100%;
    vertical-align:baseline;
    background:transparent;
}

/* change colours to suit your needs */

ins {
    background-color:#ff9;
    color:#000;
    text-decoration:none;
}

/* change colours to suit your needs */

mark {
    background-color:#ff9;
    color:#000; 
    font-style:italic;
    font-weight:bold;
}

del {
    text-decoration: line-through;
}

abbr[title], dfn[title] {
    border-bottom:1px dotted;
    cursor:help;
}

table {
    border-collapse:collapse;
    border-spacing:0;
}

/* change border colour to suit your needs */

hr {
    display:block;
    height:1px;
    border:0;   
    border-top:1px solid #cccccc;
    margin:1em 0;
    padding:0;
}

input, select {
    vertical-align:middle;
}

h1 {
    color: #000000;
    size: 32px;
}"
`;

exports[`Client gets bundled js file from build server 1`] = `
"(function(){function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require==\\"function\\"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error(\\"Cannot find module '\\"+o+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require==\\"function\\"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s}return e})()({1:[function(require,module,exports){
'use strict';

const { hello } = require('./lib');

console.log(hello());

},{\\"./lib\\":2}],2:[function(require,module,exports){
'use strict';

module.exports.hello = function hello() {
    console.log('hello');
};

},{}]},{},[1]);
"
`;

exports[`Client gets uploaded css feed from build server 1`] = `
Array [
  Object {
    "content": "html, body, div, span, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
abbr, address, cite, code,
del, dfn, em, img, ins, kbd, q, samp,
small, strong, sub, sup, var,
b, i,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section, summary,
time, mark, audio, video {
    margin:0;
    padding:0;
    border:0;
    outline:0;
    font-size:100%;
    vertical-align:baseline;
    background:transparent;
}

body {
    line-height:1;
}

article,aside,details,figcaption,figure,
footer,header,hgroup,menu,nav,section { 
    display:block;
}

nav ul {
    list-style:none;
}

blockquote, q {
    quotes:none;
}

blockquote:before, blockquote:after,
q:before, q:after {
    content:'';
    content:none;
}

a {
    margin:0;
    padding:0;
    font-size:100%;
    vertical-align:baseline;
    background:transparent;
}

/* change colours to suit your needs */

ins {
    background-color:#ff9;
    color:#000;
    text-decoration:none;
}

/* change colours to suit your needs */

mark {
    background-color:#ff9;
    color:#000; 
    font-style:italic;
    font-weight:bold;
}

del {
    text-decoration: line-through;
}

abbr[title], dfn[title] {
    border-bottom:1px dotted;
    cursor:help;
}

table {
    border-collapse:collapse;
    border-spacing:0;
}

/* change border colour to suit your needs */

hr {
    display:block;
    height:1px;
    border:0;   
    border-top:1px solid #cccccc;
    margin:1em 0;
    padding:0;
}

input, select {
    vertical-align:middle;
}

h1 {
    color: #000000;
    size: 32px;
}",
    "file": "integration/assets/style.css",
    "id": "8b30de6afdb4474cfd5c5df052408dc6ee170d243861c3fa63d246440b5f472d",
    "name": "integration",
    "version": "1.0.0",
  },
]
`;

exports[`Client gets uploaded js feed from build server 1`] = `
Array [
  Object {
    "deps": Object {
      "./lib": "756bd25204025b80a24529f1c61b9ee2869064d92c8ce001f352150effecb172",
    },
    "entry": true,
    "expose": false,
    "file": "integration/assets/index.js",
    "id": "9a92a78fe0549160c6de69e707ffb3ce54aa13654019a014d6d6e0144f59b1c7",
    "index": 1,
    "indexDeps": Object {
      "./lib": 2,
    },
    "order": 0,
    "source": "'use strict';

const { hello } = require('./lib');

console.log(hello());
",
  },
  Object {
    "deps": Object {},
    "file": "integration/assets/lib.js",
    "id": "756bd25204025b80a24529f1c61b9ee2869064d92c8ce001f352150effecb172",
    "index": 2,
    "indexDeps": Object {},
    "source": "'use strict';

module.exports.hello = function hello() {
    console.log('hello');
};
",
  },
]
`;

exports[`Client requests bundling of a css feed on build server 1`] = `
Object {
  "file": "8b30de6afdb4474cfd5c5df052408dc6ee170d243861c3fa63d246440b5f472d.css",
  "uri": "http://127.0.0.1:7100/bundle/8b30de6afdb4474cfd5c5df052408dc6ee170d243861c3fa63d246440b5f472d.css",
}
`;

exports[`Client requests bundling of a js feed on build server 1`] = `
Object {
  "file": "97d7333aa39161bba7376398bd3527851539260d715e3bbd2a6f4e4e67217f8c.js",
  "uri": "http://127.0.0.1:7100/bundle/97d7333aa39161bba7376398bd3527851539260d715e3bbd2a6f4e4e67217f8c.js",
}
`;

exports[`Client uploads a css feed to build server 1`] = `
Object {
  "file": "80c8a092aee329dd8cf852eaa081829bcac6f24bfa4ef11b90c94b9607244251.json",
  "id": "80c8a092aee329dd8cf852eaa081829bcac6f24bfa4ef11b90c94b9607244251",
  "uri": "http://127.0.0.1:7100/feed/80c8a092aee329dd8cf852eaa081829bcac6f24bfa4ef11b90c94b9607244251.json",
}
`;

exports[`Client uploads a js feed to build server 1`] = `
Object {
  "file": "a7fb04f1606e30f96517f844e45231be214d9e8d6547f4237e5a88348659f993.json",
  "id": "a7fb04f1606e30f96517f844e45231be214d9e8d6547f4237e5a88348659f993",
  "uri": "http://127.0.0.1:7100/feed/a7fb04f1606e30f96517f844e45231be214d9e8d6547f4237e5a88348659f993.json",
}
`;

exports[`NODE_ENV variables replaced with hard coded values 1`] = `
"(function() {
  function e(t, n, r) {
    function s(o, u) {
      if (!n[o]) {
        if (!t[o]) {
          var a = typeof require == \\"function\\" && require;
          if (!u && a) return a(o, !0);
          if (i) return i(o, !0);
          var f = new Error(\\"Cannot find module '\\" + o + \\"'\\");
          throw ((f.code = \\"MODULE_NOT_FOUND\\"), f);
        }
        var l = (n[o] = { exports: {} });
        t[o][0].call(
          l.exports,
          function(e) {
            var n = t[o][1][e];
            return s(n ? n : e);
          },
          l,
          l.exports,
          e,
          t,
          n,
          r
        );
      }
      return n[o].exports;
    }
    var i = typeof require == \\"function\\" && require;
    for (var o = 0; o < r.length; o++) s(r[o]);
    return s;
  }
  return e;
})()(
  {
    1: [
      function(require, module, exports) {
        (function(process) {
          if (\\"development\\" === \\"production\\") {
            console.log(\\"In prod!!\\");
          } else if (\\"development\\" === \\"development\\") {
            console.log(\\"In dev!!\\");
          } else {
            console.log(\\"In limbo!!\\");
          }
        }.call(this, require(\\"./../node_modules/process/browser.js\\")));
      },
      { \\"./../node_modules/process/browser.js\\": 2 }
    ],
    2: [
      function(require, module, exports) {
        // shim for using process in browser
        var process = (module.exports = {});

        // cached from whatever global is present so that test runners that stub it
        // don't break things.  But we need to wrap it in a try catch in case it is
        // wrapped in strict mode code which doesn't define any globals.  It's inside a
        // function because try/catches deoptimize in certain engines.

        var cachedSetTimeout;
        var cachedClearTimeout;

        function defaultSetTimout() {
          throw new Error(\\"setTimeout has not been defined\\");
        }
        function defaultClearTimeout() {
          throw new Error(\\"clearTimeout has not been defined\\");
        }
        (function() {
          try {
            if (typeof setTimeout === \\"function\\") {
              cachedSetTimeout = setTimeout;
            } else {
              cachedSetTimeout = defaultSetTimout;
            }
          } catch (e) {
            cachedSetTimeout = defaultSetTimout;
          }
          try {
            if (typeof clearTimeout === \\"function\\") {
              cachedClearTimeout = clearTimeout;
            } else {
              cachedClearTimeout = defaultClearTimeout;
            }
          } catch (e) {
            cachedClearTimeout = defaultClearTimeout;
          }
        })();
        function runTimeout(fun) {
          if (cachedSetTimeout === setTimeout) {
            //normal enviroments in sane situations
            return setTimeout(fun, 0);
          }
          // if setTimeout wasn't available but was latter defined
          if (
            (cachedSetTimeout === defaultSetTimout || !cachedSetTimeout) &&
            setTimeout
          ) {
            cachedSetTimeout = setTimeout;
            return setTimeout(fun, 0);
          }
          try {
            // when when somebody has screwed with setTimeout but no I.E. maddness
            return cachedSetTimeout(fun, 0);
          } catch (e) {
            try {
              // When we are in I.E. but the script has been evaled so I.E. doesn't trust the global object when called normally
              return cachedSetTimeout.call(null, fun, 0);
            } catch (e) {
              // same as above but when it's a version of I.E. that must have the global object for 'this', hopfully our context correct otherwise it will throw a global error
              return cachedSetTimeout.call(this, fun, 0);
            }
          }
        }
        function runClearTimeout(marker) {
          if (cachedClearTimeout === clearTimeout) {
            //normal enviroments in sane situations
            return clearTimeout(marker);
          }
          // if clearTimeout wasn't available but was latter defined
          if (
            (cachedClearTimeout === defaultClearTimeout ||
              !cachedClearTimeout) &&
            clearTimeout
          ) {
            cachedClearTimeout = clearTimeout;
            return clearTimeout(marker);
          }
          try {
            // when when somebody has screwed with setTimeout but no I.E. maddness
            return cachedClearTimeout(marker);
          } catch (e) {
            try {
              // When we are in I.E. but the script has been evaled so I.E. doesn't  trust the global object when called normally
              return cachedClearTimeout.call(null, marker);
            } catch (e) {
              // same as above but when it's a version of I.E. that must have the global object for 'this', hopfully our context correct otherwise it will throw a global error.
              // Some versions of I.E. have different rules for clearTimeout vs setTimeout
              return cachedClearTimeout.call(this, marker);
            }
          }
        }
        var queue = [];
        var draining = false;
        var currentQueue;
        var queueIndex = -1;

        function cleanUpNextTick() {
          if (!draining || !currentQueue) {
            return;
          }
          draining = false;
          if (currentQueue.length) {
            queue = currentQueue.concat(queue);
          } else {
            queueIndex = -1;
          }
          if (queue.length) {
            drainQueue();
          }
        }

        function drainQueue() {
          if (draining) {
            return;
          }
          var timeout = runTimeout(cleanUpNextTick);
          draining = true;

          var len = queue.length;
          while (len) {
            currentQueue = queue;
            queue = [];
            while (++queueIndex < len) {
              if (currentQueue) {
                currentQueue[queueIndex].run();
              }
            }
            queueIndex = -1;
            len = queue.length;
          }
          currentQueue = null;
          draining = false;
          runClearTimeout(timeout);
        }

        process.nextTick = function(fun) {
          var args = new Array(arguments.length - 1);
          if (arguments.length > 1) {
            for (var i = 1; i < arguments.length; i++) {
              args[i - 1] = arguments[i];
            }
          }
          queue.push(new Item(fun, args));
          if (queue.length === 1 && !draining) {
            runTimeout(drainQueue);
          }
        };

        // v8 likes predictible objects
        function Item(fun, array) {
          this.fun = fun;
          this.array = array;
        }
        Item.prototype.run = function() {
          this.fun.apply(null, this.array);
        };
        process.title = \\"browser\\";
        process.browser = true;
        process.env = {};
        process.argv = [];
        process.version = \\"\\"; // empty string to avoid regexp issues
        process.versions = {};

        function noop() {}

        process.on = noop;
        process.addListener = noop;
        process.once = noop;
        process.off = noop;
        process.removeListener = noop;
        process.removeAllListeners = noop;
        process.emit = noop;
        process.prependListener = noop;
        process.prependOnceListener = noop;

        process.listeners = function(name) {
          return [];
        };

        process.binding = function(name) {
          throw new Error(\\"process.binding is not supported\\");
        };

        process.cwd = function() {
          return \\"/\\";
        };
        process.chdir = function(dir) {
          throw new Error(\\"process.chdir is not supported\\");
        };
        process.umask = function() {
          return 0;
        };
      },
      {}
    ]
  },
  {},
  [1]
);
"
`;

exports[`NODE_ENV variables trigger dead code elimination (DCE) 1`] = `"!function(){return function e(r,n,t){function o(u,c){if(!n[u]){if(!r[u]){var s=\\"function\\"==typeof require&&require;if(!c&&s)return s(u,!0);if(i)return i(u,!0);var f=new Error(\\"Cannot find module '\\"+u+\\"'\\");throw f.code=\\"MODULE_NOT_FOUND\\",f}var l=n[u]={exports:{}};r[u][0].call(l.exports,function(e){var n=r[u][1][e];return o(n||e)},l,l.exports,e,r,n,t)}return n[u].exports}for(var i=\\"function\\"==typeof require&&require,u=0;u<t.length;u++)o(t[u]);return o}}()({1:[function(e,r,n){(function(e){console.log(\\"In prod!!\\")}).call(this,e(\\"./../node_modules/process/browser.js\\"))},{\\"./../node_modules/process/browser.js\\":2}],2:[function(e,r,n){var t,o,i=r.exports={};function u(){throw new Error(\\"setTimeout has not been defined\\")}function c(){throw new Error(\\"clearTimeout has not been defined\\")}function s(e){if(t===setTimeout)return setTimeout(e,0);if((t===u||!t)&&setTimeout)return t=setTimeout,setTimeout(e,0);try{return t(e,0)}catch(r){try{return t.call(null,e,0)}catch(r){return t.call(this,e,0)}}}!function(){try{t=\\"function\\"==typeof setTimeout?setTimeout:u}catch(e){t=u}try{o=\\"function\\"==typeof clearTimeout?clearTimeout:c}catch(e){o=c}}();var f,l=[],a=!1,h=-1;function p(){a&&f&&(a=!1,f.length?l=f.concat(l):h=-1,l.length&&d())}function d(){if(!a){var e=s(p);a=!0;for(var r=l.length;r;){for(f=l,l=[];++h<r;)f&&f[h].run();h=-1,r=l.length}f=null,a=!1,function(e){if(o===clearTimeout)return clearTimeout(e);if((o===c||!o)&&clearTimeout)return o=clearTimeout,clearTimeout(e);try{o(e)}catch(r){try{return o.call(null,e)}catch(r){return o.call(this,e)}}}(e)}}function m(e,r){this.fun=e,this.array=r}function T(){}i.nextTick=function(e){var r=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)r[n-1]=arguments[n];l.push(new m(e,r)),1!==l.length||a||s(d)},m.prototype.run=function(){this.fun.apply(null,this.array)},i.title=\\"browser\\",i.browser=!0,i.env={},i.argv=[],i.version=\\"\\",i.versions={},i.on=T,i.addListener=T,i.once=T,i.off=T,i.removeListener=T,i.removeAllListeners=T,i.emit=T,i.prependListener=T,i.prependOnceListener=T,i.listeners=function(e){return[]},i.binding=function(e){throw new Error(\\"process.binding is not supported\\")},i.cwd=function(){return\\"/\\"},i.chdir=function(e){throw new Error(\\"process.chdir is not supported\\")},i.umask=function(){return 0}},{}]},{},[1]);"`;

exports[`Uploaded js feed from build server is not deduped 1`] = `
Array [
  Object {
    "deps": Object {
      "./d": "320f596434a079912e951a037302087837f241f2896be048e3ccd18d8e6888c9",
    },
    "file": "integration/assets/a.js",
    "id": "893576861db8f16e03815f618ca6838cb4af6fd867b10ffa5684e1f1fcd6b467",
    "index": 1,
    "indexDeps": Object {
      "./d": 2,
    },
    "source": "'use strict';

const sum = require('./d');

sum(1, 2);
",
  },
  Object {
    "deps": Object {},
    "file": "integration/assets/d.js",
    "id": "320f596434a079912e951a037302087837f241f2896be048e3ccd18d8e6888c9",
    "index": 2,
    "indexDeps": Object {},
    "source": "'use strict';

module.exports = function sum(a, b) {
    return a + b;
};
",
  },
  Object {
    "deps": Object {
      "./a": "893576861db8f16e03815f618ca6838cb4af6fd867b10ffa5684e1f1fcd6b467",
    },
    "file": "integration/assets/dup-1.js",
    "id": "c2d61953741441db400b08c0432c6cc3fafcf5e8073d15ac9aa30f443718819d",
    "index": 3,
    "indexDeps": Object {
      "./a": 1,
    },
    "source": "require('./a');
",
  },
  Object {
    "deps": Object {
      "./a": "893576861db8f16e03815f618ca6838cb4af6fd867b10ffa5684e1f1fcd6b467",
    },
    "file": "integration/assets/dup-2.js",
    "id": "3b988745ee60ea0d85d16f6e70684ddef88e13d39ce1be5e638af1dd78018aba",
    "index": 4,
    "indexDeps": Object {
      "./a": 1,
    },
    "source": "require('./a');
",
  },
  Object {
    "deps": Object {
      "./a": "893576861db8f16e03815f618ca6838cb4af6fd867b10ffa5684e1f1fcd6b467",
    },
    "file": "integration/assets/dup-3.js",
    "id": "9851e36fb1b5aed80b4d16a30a36c8a2872b9a1abd32b47ee3e1d5c4378c09bc",
    "index": 5,
    "indexDeps": Object {
      "./a": 1,
    },
    "source": "require('./a');
",
  },
  Object {
    "deps": Object {
      "./dup-1": "c2d61953741441db400b08c0432c6cc3fafcf5e8073d15ac9aa30f443718819d",
      "./dup-2": "3b988745ee60ea0d85d16f6e70684ddef88e13d39ce1be5e638af1dd78018aba",
      "./dup-3": "9851e36fb1b5aed80b4d16a30a36c8a2872b9a1abd32b47ee3e1d5c4378c09bc",
    },
    "entry": true,
    "expose": false,
    "file": "integration/assets/duped-before-uploading.js",
    "id": "e1191d059ff0923159cbb3625c31de6f18ea5eea053844442512d8c86c6d918e",
    "index": 6,
    "indexDeps": Object {
      "./dup-1": 3,
      "./dup-2": 4,
      "./dup-3": 5,
    },
    "order": 0,
    "source": "const a = require('./dup-1');
const b = require('./dup-2');
const c = require('./dup-3');
",
  },
]
`;
